src/prototype_room_init.js

```diff
@@ -63,6 +63,42 @@ Room.prototype.setTowerFiller = function() {
   }
 };
 
+function setLabsTerminal(room, path, costMatrixBase) {
+  for (let pathI = path.length - 1; pathI > 0; pathI--) {
+    let pathPos = new RoomPosition(path[pathI].x, path[pathI].y, room.name);
+    let structurePosIterator = pathPos.findNearPosition();
+    for (let structurePos of structurePosIterator) {
+      if (room.memory.position.structure.lab.length < CONTROLLER_STRUCTURES.lab[8]) {
+        room.memory.position.structure.lab.push(structurePos);
+        costMatrixBase.set(structurePos.x, structurePos.y, config.layout.structureAvoid);
+        continue;
+      }
+      if (room.memory.position.structure.terminal.length < CONTROLLER_STRUCTURES.terminal[8]) {
+        room.memory.position.structure.terminal.push(structurePos);
+        costMatrixBase.set(structurePos.x, structurePos.y, config.layout.structureAvoid);
+        room.memory.position.pathEnd = [pathPos];
+        continue;
+      }
+      if (room.memory.position.structure.lab.length < CONTROLLER_STRUCTURES.lab[8] ||
+        room.memory.position.structure.terminal.length < CONTROLLER_STRUCTURES.terminal[8]) {
+        room.log('Structures not found: ' +
+          'lab: ' + room.memory.position.structure.lab.length + ' ' +
+          'terminal: ' + room.memory.position.structure.terminal.length
+        );
+        continue;
+      }
+      if (!room.memory.position.pathEnd) {
+        room.log('Room not completly build');
```

Comment:
You will always get this error, because `setStructures` is called after this and `pathEnd` is set there